dev_mode/index.js

```diff
@@ -175,6 +176,23 @@ export async function main() {
 
       const pkgPlugins = pkgJson['jupyterlab']['plugins'];
       if (pkgPlugins) {
+        const entryPointData = pkgJson['jupyterlab']['entrypoint'];
+
+        if(entryPointData){
+          for(const widgetFactory of entryPointData.widgetFactory){
+            entrypoints.push({
+              extension: '{{@key}}',
+              data : {
+                name: widgetFactory.name,
+                fileTypes: widgetFactory.fileTypes,
+                defaultFor: widgetFactory.defaultFor
+              },
+              activate: async ()=>{
+                return pluginRegistry.activatePlugin(widgetFactory.pluginId)
+              }
+            })
+          }
```

Comment:
This is too specific. We should generalize it with something like:

```suggestion
          Object.entries(entryPointData).forEach(([entryPoint, data]) => {
            if (!entrypoints[entryPoint]) {
              entrypoints[entryPoint] = []
            }
            entrypoints[entryPoint].push({
              extension: '{{@key}}',
              data,
              activate: () => {
                return pluginRegistry.activatePlugin(pluginId)
              }
            })
          });
```